metrics/time/launch_times.sh

```diff
@@ -0,0 +1,241 @@
+#!/bin/bash
+# Copyright (c) 2017-2018 Intel Corporation
+# 
+# SPDX-License-Identifier: Apache-2.0
+#
+#  Description of the test:
+#  This test takes a number of time measurements through the complete
+#  launch/shutdown cycle of a single container.
+#  From those measurements it derives a number of time measures, such as:
+#   - time to payload execution
+#   - time to get to VM kernel
+#   - time in VM kernel boot
+#   - time to quit
+#   - total time (from launch to finished)
+#
+# Note, the <image> used for this test must support the full 'date' command
+# syntax - the date from busybox for isntance *does not* support this, so
+# will not work with this test.
+#
+# Note, this test launches a single container at a time, that quits - thus,
+# this test measures times for the 'first container' only. This test does
+# not look for any scalability slowdowns as the number of running containers
+# increases for instance - that is handled in other tests
+
+set -e
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+# Calculating the kernel time from dmesg stamps only really works for VM
+# based runtimes - we dynamically enable it if we find we are using a known
+# VM runtime
+CALCULATE_KERNEL=
+
+REQUIRED_CMDS=("bc" "awk")
+
+# The date command format we use to ensure we capture the ns timings
+# Note the no-0-padding - 0 padding the results breaks bc in some cases
+DATECMD="date -u +%-s:%-N"
+
+
+# convert a 'seconds:nanoseconds' string into nanoseconds
+sn_to_ns() {
```

Comment:
not sure if this could be relevant to other tests? and if so, could this be added in the `lib/common.bash` ?

Reply:
Maybe. It is very close/similar to the timestamp code we have in the json.sh in the lib already, but subtly different enough at the moment that I don't think we can directly re-use.
Let's keep an eye out for any future tests that need to do timestamp collection/processing/diffs, and if we find a case then let's try to re-factor at that point.